src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hosting/VSInternalServerCapabilitiesExtensions.cs

```diff
@@ -40,6 +40,19 @@ public static SemanticTokensOptions EnableSemanticTokens(this SemanticTokensOpti
         return options;
     }
 
+    public static void EnableSignatureHelp(this VSInternalServerCapabilities serverCapabilities)
+    {
+        serverCapabilities.SignatureHelpProvider = new SignatureHelpOptions().EnableSignatureHelp();
+    }
+
+    public static SignatureHelpOptions EnableSignatureHelp(this SignatureHelpOptions options)
+    {
+        options.TriggerCharacters = ["(", ",", "<"];
+        options.RetriggerCharacters = [">", ")"];
```

Comment:
As a super-nit - seems like this is very removed from the actual signature help request code. I personally would've preferred to keep it close to the consuming code rather than in centralized location. I don't have strong feeling about it though :) feel free to punt of course.

Reply:
I understand where you've coming from, but the only other option here would be to duplicate this info, and I don't like that. Also, we don't have any "consuming code". We have two endpoints, and they both need to declare the right capabilities, and the client is the only thing that uses the info. If we actually needed to use this info then it might make sense to put it on a service or something, but signature help is extremely simple, and we do absolutely no processing of requests or responses, essentially.

In future if we remove the "normal" LSP endpoints then these can all move to be part of the cohosting endpoints I guess.